src/controller.rs

```diff
@@ -243,18 +249,33 @@ pub fn grab_pan(
 pub fn rotate(
     mut cam_q: Query<(&mut RtsCamera, &RtsCameraControls)>,
     mouse_input: Res<ButtonInput<MouseButton>>,
+    keys: Res<ButtonInput<KeyCode>>,
     mut mouse_motion: EventReader<MouseMotion>,
     primary_window_q: Query<&Window, With<PrimaryWindow>>,
 ) {
-    for (mut cam, controller) in cam_q.iter_mut().filter(|(_, ctrl)| ctrl.enabled) {
-        if mouse_input.pressed(controller.button_rotate) {
-            let mouse_delta = mouse_motion.read().map(|e| e.delta).sum::<Vec2>();
-            if let Ok(primary_window) = primary_window_q.get_single() {
+    if let Ok(primary_window) = primary_window_q.get_single() {
+        for (mut cam, controller) in cam_q.iter_mut().filter(|(_, ctrl)| ctrl.enabled) {
+            if mouse_input.pressed(controller.button_rotate) {
+                let mouse_delta = mouse_motion.read().map(|e| e.delta).sum::<Vec2>();
                 // Adjust based on window size, so that moving mouse entire width of window
                 // will be one half rotation (180 degrees)
                 let delta_x = mouse_delta.x / primary_window.width() * PI;
                 cam.target_focus.rotate_local_y(-delta_x);
             }
+
+            cam.target_focus.rotate_local_y(
+                (if keys.pressed(controller.key_rotate_right) {
+                    1.0
+                } else {
+                    0.0
+                } - if keys.pressed(controller.key_rotate_left) {
+                    1.0
+                } else {
+                    0.0
+                }) / primary_window.width()
+                    * PI
+                    * 16.0,
+            );
```

Comment:
This is very hard to read. Could you please use some variables?

Also, I think it makes sense to only allow rotation by mouse OR keyboard (similar to panning). So let's prevent mouse rotation if one of the rotation keys is pressed.